lib/chef/provider/git.rb

```diff
@@ -41,6 +41,20 @@ def load_current_resource
       end
 
       def define_resource_requirements
+        unless new_resource.user.nil?
```

Comment:
Need to consider `new_resource.user` is nil as well for the assertion so that validate properly

Reply:
new_resource.user is an optional property and the setting of the env var below is gated so this seems fine to me.